tests/test_functional.py

```diff
@@ -287,18 +287,119 @@ def test_logout_no_confirmation(client):
-    # FIXME: currently not working
-    pass
+    """Logout without confirmation requires the id_token"""
+
+    user = UserFactory()
+    ApplicationFactory(client_id=OIDC_PARAMS["client_id"])
+
+    auth_url = reverse("oidc_overrides:authorize")
+    auth_complete_url = add_url_params(auth_url, OIDC_PARAMS)
+    response = client.get(auth_complete_url)
+    assertRedirects(response, reverse("accounts:login"))
+
+    response = client.post(response.url, data={"email": user.email, "password": DEFAULT_PASSWORD})
+    assert get_user(client).is_authenticated is True
+    response = client.get(response.url)
+    auth_response_params = get_url_params(response.url)
+    oidc_flow_followup(client, auth_response_params, user)
+
+    assert get_user(client).is_authenticated is True
+    response = call_logout(
+        client, "get", {"client_id": OIDC_PARAMS["client_id"], "post_logout_redirect_uri": "http://callback/"}
+    )
+    assert response.status_code == 200
+
+    response = call_logout(
+        client,
+        "post",
+        {"client_id": OIDC_PARAMS["client_id"], "post_logout_redirect_uri": "http://callback/", "allow": True},
+    )
+    assertRedirects(response, "http://callback/", fetch_redirect_response=False)
+    assert not get_user(client).is_authenticated
+    assert token_are_revoked(user)
+
+    response = client.get(auth_complete_url)
+    assertRedirects(response, reverse("accounts:login"))
+
+
+def test_logout_with_confirmation_when_already_session_and_tokens_already_expired_with_client_id(client):
+    """Logout without confirmation requires the id_token"""
+
+    user = UserFactory()
+    ApplicationFactory(client_id=OIDC_PARAMS["client_id"])
+
+    with freeze_time("2023-05-25 9:34"):
+        auth_url = reverse("oidc_overrides:authorize")
+        auth_complete_url = add_url_params(auth_url, OIDC_PARAMS)
+        response = client.get(auth_complete_url)
+        assertRedirects(response, reverse("accounts:login"))
+
+        response = client.post(response.url, data={"email": user.email, "password": DEFAULT_PASSWORD})
+        assert get_user(client).is_authenticated is True
+        response = client.get(response.url)
+        auth_response_params = get_url_params(response.url)
+        oidc_flow_followup(client, auth_response_params, user)
+
+    with freeze_time("2023-05-25 20:05"):
+        assert get_user(client).is_authenticated is False
+        response = call_logout(
+            client, "get", {"client_id": OIDC_PARAMS["client_id"], "post_logout_redirect_uri": "http://callback/"}
+        )
+        assert response.status_code == 200
+
+        response = call_logout(
+            client,
+            "post",
+            {"client_id": OIDC_PARAMS["client_id"], "post_logout_redirect_uri": "http://callback/", "allow": True},
+        )
+
+        assertRedirects(response, "http://callback/", fetch_redirect_response=False)
+        assert not token_are_revoked(user)  # Unable to revoke tokens
```

Comment:
Ce comportement m’attriste. L’utilisateur vient de se déconnecter, si le partenaire ne nous donne pas l’identité de l’utilisateur, on ne va juste rien faire. À quoi bon rediriger vers IC ?

Mais la [spec](https://openid.net/specs/openid-connect-rpinitiated-1_0.html) dit bien :
> Note that because RP-Initiated Logout Requests are intended to be idempotent, it is explicitly not an error for an RP to request that a logout be performed when the OP does not consider that the End-User is logged in with the OP at the requesting RP. 

```suggestion
        # The user is anonymous, without the `id_token`, the system cannot identify the user.
        # Without the user, their tokens cannot be revoked.
        assert not token_are_revoked(user)
```